scalar/struct.go

```diff
@@ -0,0 +1,74 @@
+package scalar
+
+import (
+	"encoding/json"
+	"reflect"
+
+	"github.com/apache/arrow/go/v13/arrow"
+)
+
+type Struct struct {
+	Valid bool
+	Value any
+
+	Type arrow.DataType
+}
+
+func (s *Struct) IsValid() bool {
+	return s.Valid
+}
+
+func (s *Struct) Equal(rhs Scalar) bool {
+	if rhs == nil {
+		return false
+	}
+	r, ok := rhs.(*Struct)
+	if !ok {
+		return false
+	}
+	return s.Valid == r.Valid && reflect.DeepEqual(s.Value, r.Value)
+}
+
+func (s *Struct) String() string {
+	if !s.Valid {
+		return nullValueStr
+	}
+	return s.Type.String() + " value"
```

Comment:
do we need `+ " value"` & why?

Reply:
`String()` is supposed to stringify the value (for debugging? I don't know why.. since everything else is supposed to use `Get() any`) the +value makes it apparent that it's a (fake) value and not the type we're printing.

Reply:
IMO the scalar.String just returns the value that we could use in the future for some purposes (e.g., append to `*array.StringBuilder`).
As other scalars show a proper value, I think these should show it, too.
cc: @yevgenypats @hermanschaaf

Reply:
added json.Marshal instead of this